packages/amino/src/encoding.ts

```diff
@@ -77,6 +82,53 @@ export function decodeBech32Pubkey(bechEncoded: string): Pubkey {
   return decodeAminoPubkey(data);
 }
 
+/**
+ * Decodes a multisig pubkey to type object.
+ * Pubkey structure [ prefix + const + threshold + loop:(const + pubkeyLength + pubkey            ) ]
+ *                  [   4b   + 1b    +   1b      + loop:(1b    +    1b        + pubkeyLength bytes) ]
+ * @param data encoded pubkey
+ */
+function decodeMultisigPubkey(data: Uint8Array): MultisigThresholdPubkey {
+  if (data.length < 4 + 1 + 1) {
+    // verify that we can read at least threshold
+    throw new Error("Invalid multisig data length.");
+  }
+  let rest = data.slice(pubkeyAminoPrefixSecp256k1.length);
```

Comment:
The way you use `rest` here feels like it would be better to convert the `Uint8Array` to a mutable array: `Array.from(data)`. Then you can use `shift` and `splice` to get and delete in one call.

Reply:
Golang developer detected :)